### ISSUE TITLE
Consuming a simple ONNX model for text classification from ML.Net

### ISSUE THREAD

**Issue by oliverw** [onnx/sklearn-onnx#265]
I'm trying to consume a trained scikit learn model that was converted to ONNX from ML.Net but I'm not sure if this is actually an issue with ML.Net, sklearn-onnx or my model. Attempting to load the onnx model in ML.Net results in the following error:

> System.ArgumentOutOfRangeException: 'dim (Parameter 'Dimension { } in ONNX tensor cannot exceed the maximum of 32-bit signed integer.')
> Actual value was 0.'
> 

Inspecting the generated model in Netron made me curious whether the error might be triggered by both outputs having a zero length?

![image](https://user-images.githubusercontent.com/51688/65391096-6fe19180-dd65-11e9-965a-3eab334533a6.png)

**Model creation in python:**

```python
import os
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.svm import LinearSVC
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import StringTensorType
import pandas as pd
from onnxmltools import save_model

data = pd.read_csv('training_data.csv', sep='\t')

pipeline = Pipeline([
  ('tfidf', TfidfVectorizer()),
  ('clf', LinearSVC()),
])

pipeline.fit(data["Keywords"], data["Label"])

onnx = convert_sklearn(pipeline, name='test_predictor',
  initial_types=[('input', StringTensorType([1, 1]))])

save_model(onnx, "test.onnx")
```

**Comment by prabhat00155** [comment]
Which version of sklearn-onnx and onnxruntime are you using? Can you do a pip freeze | grep onnx?
$ pip freeze | grep onnx
keras2onnx==1.4.0
onnx==1.5.0
onnxconverter-common==1.5.2
onnxmltools==1.5.0
onnxruntime==0.5.0
skl2onnx==1.5.1

I tried the aforementioned script on a sample dataset and was able to run it successfully. However, I see some differences in the onnx models itself.

![image](https://user-images.githubusercontent.com/7043157/65555141-98af8580-df23-11e9-8389-492de5da09fb.png)

**Comment by oliverw** [comment]
> keras2onnx==1.5.1
> onnx==1.5.0
> onnxconverter-common==1.5.3
> onnxmltools==1.5.0
> onnxruntime==0.5.0
> skl2onnx==1.5.1

> Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)] on **win32**
> Type "help", "copyright", "credits" or "license" for more information.

**Comment by prabhat00155** [comment]
Can you try running the following in Jupyter and see if it runs? If yes, can you paste the error you see:

import numpy as np
from sklearn.datasets import  fetch_20newsgroups
from sklearn.model_selection import train_test_split
from sklearn.pipeline import Pipeline
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.svm import LinearSVC
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType, Int64TensorType
from onnxruntime import InferenceSession
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import StringTensorType
from onnxmltools import save_model

data = fetch_20newsgroups()
X, y = np.array(data.data), data.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=42)
pipeline = Pipeline([
  ('tfidf', TfidfVectorizer()),
  ('clf', LinearSVC()),
])
pipeline.fit(X_train, y_train)
onnx = convert_sklearn(pipeline, name='test_predictor',
  initial_types=[('input', StringTensorType([1, 1]))])

save_model(onnx, "test.onnx")
sess = InferenceSession('test.onnx')
sess.run('', input_feed={'input': X_test})

Also, when did you see the error you have reported above? Is it InferenceSession()?

**Comment by prabhat00155** [comment]
Hang on, I think I missed the first part. Are you able to run the onnx model in Python(using onnxruntime) and getting error only with ml.net? In that case, I think there is an issue with ml.net.

**Comment by oliverw** [comment]
Yes the model works fine in Python. Just ML.Net's ONNX loader chokes on it.

**Comment by prabhat00155** [comment]
@xadupre: Do you know why the type is string[0]?

**Comment by xadupre** [comment]
Maybe that's a way to indicate that the model is supporting broadcasting and can predict batches. Otherwise I don't know.
